types/react-select/index.d.ts

```diff
@@ -346,7 +346,7 @@ declare namespace ReactSelectClass {
         /**
          * option component to render in dropdown
          */
-        optionComponent?: React.ComponentType;
+        optionComponent?: JSX.Element;
```

Comment:
[`React.ComponentType`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2e8fc9bb9820a7a03e96a9658544f2bf81d91e3b/types/react/index.d.ts#L72) does exist and is different from `JSX.Element`, which is a shortcut for [`React.ReactElement<any>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2e8fc9bb9820a7a03e96a9658544f2bf81d91e3b/types/react/index.d.ts#L3451).

A `React.ComponentType` is a react component while a `JSX.Element` is what a react component returns:

```ts
const MyComponent: React.SFC = (): JSX.Element =>
   <div>Hello World</div>;
```
In the above example, `MyComponent` can be passed as a `React.ComponentType`, while `<div>Hello World</div>` (or `<MyComponent />`) would be a `JSX.Element`.